### PR TITLE
Fix learning bugs and switch to MNIST dataset

### PR DESCRIPTION
**Pull request by maharshmellow**
This PR fixes some of the bugs associated with our learning implementation and also switches to the MNIST dataset.

### CODE REVIEW

**Comment by crankycoder** [comment]
@maharshmellow @zhaoyuxuan  what are we waiting on to close off the WIP flag?

**Comment by maharshmellow** [comment]
@crankycoder Just had to update with master 😄

**Comment by crankycoder** [comment]
@maharshmellow Sweet!  I'll try to take a look at this today.

File: mozfldp/simulation_util.py [hunk 1]
```diff
@@ -149,10 +151,10 @@ def server_update(
             print("Updated Weights: ", coef, intercept)
 
     # load coefficients and intercept into the classifier
-    clf = SGDClassifier(loss="hinge", penalty="l2")
+    clf = SGDClassifier(loss="log")
```
Comment (crankycoder) [inline mozfldp/simulation_util.py:154]: Oh - did the hinge loss cause problems here?

**Comment by crankycoder** [comment]
@maharshmellow I get a crash early in the simulation in the second cell of the `Federated_Learning_Simulation.ipynb`

Can you verify you're not getting the same crash?

image

File: mozfldp/simulation_util.py [hunk 2]
```diff
@@ -23,23 +23,25 @@ def client_update(init_weights, epochs, batch_size, features, labels):
...
-    # set max_iter to 1 so that each .fit() call only does one training step
-    classifier = SGDClassifier(loss="hinge", penalty="l2", max_iter=1)
+    classifier = SGDClassifier(loss="log")
+    classifier.coef_ = np.array(coef)
+    classifier.intercept_ = np.array(intercept)
 
     for epoch in range(epochs):
         for i in range(len(batches_features)):
-            classifier.fit(
+            classifier.partial_fit(
```
Comment (crankycoder) [inline mozfldp/simulation_util.py:40]: Just for my clarity - is 0-9 the complete list of all labels available in the MNIST dataset?

**Comment by crankycoder** [comment]
Let's pull out the bug with the one cell into a separate PR so we can simultaneously get testcases running while the ipynb is being fixed.

**Pull request merged**